lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -361,8 +361,23 @@ static bool isStoreCopy(SILValue value) {
     SmallVector<SILValue, 4> roots;
     findGuaranteedReferenceRoots(source, /*lookThroughNestedBorrows=*/true,
                                  roots);
-    if (llvm::any_of(roots, [](SILValue root) {
-          return isa<BeginApplyInst>(root->getDefiningInstruction());
+    SSAPrunedLiveness liveness;
+    if (llvm::any_of(roots, [&liveness, storeInst](SILValue root) {
+          if (isa<BeginApplyInst>(root->getDefiningInstruction())) {
+            return true;
+          }
+          liveness.initializeDef(root);
+          auto summary = liveness.computeSimple();
```

Comment:
This seems reasonably safe. But in the future we should be more precise and efficient by doing
- visitBorrowIntroducers (instead of roots)
- ExtendedLiveness.compute for each introducer (I haven't posted the PR yet)

That would either find a guaranteed argument and do no work at all, or find a begin_borrow and only visit the end_borrows. It could also handle reborrow-extended liveness.

Up to you if you want a TODO here

Reply:
Sounds great!  Added a TODO.